DDF-3822 Add configuration for timezone and time format within Intrigue

JGatley opened the pull request:
#### What does this PR do?

This PR:

- Allows user configuration in Intrigue to select a timezone other than local time to display.
- Adds user configuration in Intrigue to now display ISO 8601 time in addition to the previous 12 and 24 hour displays.
- Allows user to manually specify a timezone during temporal search via the format "**(+/-)XX:XX**". Note that currently, these times will be converted back into local time. This addition does not effect the way that temporal interacts with the catalog. 

#### Who is reviewing it? 
@jefchng 
@emmberk 
@clockard 
@oconnormi 
@kcover 
#### Select relevant component teams: 
@codice/ui 
#### Choose 2 committers to review/merge the PR. 
@andrewkfiedler
@bdeining
#### How should this be tested? (List steps with links to updated documentation)
Build the UI module.
Open Intrigue and ingest (any) product.
Verify that the time format options properly update the example time as well as the times listed on the metacard.
Verify that the timezone options properly update the example time as well as the times listed on the metacard.
Verify that temporal search works for the default timezone. (System's local time)
Verify that specifying a timezone other than local will execute a proper temporal search. (The time in the datepicker may be immediately adjusted to local time.)
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-3822](https://codice.atlassian.net/browse/DDF-3822)
#### Screenshots (if appropriate)

<img width="1272" alt="screen shot 2018-05-08 at 9 03 29 am" src="https://user-images.githubusercontent.com/9065952/39758955-5aa2f446-529f-11e8-9809-31f0ae9e147b.png">

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

JGatley commented:
Addressed Codacy findings and edited documentation to reflect changes made by this commit.

JGatley commented:
build now

JGatley commented:
build now

JGatley commented:
build now

andrewkfiedler commented:
Everything looks fine except for the immediately conversion that's happening with the datetimepicker.  I spent a little time looking into it and was able to find a fix which I pushed up in a commit here: https://github.com/andrewkfiedler/ddf/commit/4b204e24f179f53af0e1938cdcbbfd499116c934

It appears there is an undocumented timeZone option that we can pass in to the datepicker library (https://github.com/Eonasdan/bootstrap-datetimepicker/issues/2118).  

The other thing I had to do was update the yarn lock so that the library would use the same moment as everything else in Intrigue so that we could properly shim in moment-timezone.  Without it, the library doesn't get moment-timezone shimmed in and thus ignores the timeZone option (see https://github.com/Eonasdan/bootstrap-datetimepicker/blob/master/src/js/bootstrap-datetimepicker.js#L137).

andrewkfiedler commented on ui/packages/catalog-ui-search/src/main/webapp/component/time-settings/time-settings.view.js:
```diff
@@ -55,18 +57,110 @@ module.exports = Marionette.LayoutView.extend({
+                label: 'Time Zone',
+                value: [timeZone],
+                enum: [{
+                    label: 'UTC, +00:00',
```
Might be worth pushing this to the middle or having a duplicate there, I was a bit surprised during testing when looking for `+00:00` between `-01:00` and `+01:00`.

JGatley commented:
Addressed @andrewkfiedler 's comment about the TimeZone list by moving the UTC button in between -01:00 and +01:00. Squashed his datepicker timezone fix into my commit and briefly tested that temporal is still returning proper results based on the timezone selected. 

build now

emmberk commented:
A successful CI build on this PR is blocked by https://github.com/codice/ddf/pull/3231

JGatley commented:
build now

emmberk commented:
build now
